src/main/java/org/jabref/logic/util/io/FileNameHandler.java

```diff
@@ -67,7 +71,41 @@ public static String cleanDirectoryName(String badFileName) {
         return cleanName.toString().trim();
     }
 
+    /**
+     * Returns an absolute path to a file which does not match with any existing file names
+     *
+     * @param targetDirectory The directory in which file name should be unique
+     * @param fileName Suggested name for the file
+     * @return a file-name such that it does not match any existing files in targetDirectory.
+     * */
+    public static String getNonOverWritingFileName(Path targetDirectory, String fileName) {
+        String absoluteName = targetDirectory.resolve(fileName)
+                                            .toString();
+
+        Optional<String> extensionOptional = FileUtil.getFileExtension(fileName);
+
+        // the suffix include the '.' , if extension is present Eg: ".pdf"
+        String extensionSuffix = "";
```

Comment:
Please put that into an else path in the if of line 91. (Declaring the variables here is OK). I find it hard if the contents of the variables are overwritten: The default case is that there is an extension.